src/Razor/Microsoft.AspNetCore.Razor.Language/src/Extensions/ViewCssScopePass.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.Extensions
+{
+    internal class ViewCssScopePass : IntermediateNodePassBase, IRazorOptimizationPass
```

Comment:
In terms of code-organization, MVC specific types have typically existed in the .Extensions assembly. In addition, we probably want to put this behind a language version check (we don't want this running on older versions of MVC).

(cc @NTaylorMullen)

Reply:
That's a fair point, but I don't think this is MVC specific, this is specifc to cshtml. Also, if that's the case, why is the components version of this (as well as A LOT of other stuff in this assembly instead of on its own Components assembly?)